### PR TITLE
Add support for elevation requirement in COM

### PR DESCRIPTION
**Pull request by ryfu-msft**
Changes: 
- Modify PackageManager.idl to add support for ElevationRequirement
 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://api.github.com/repos/microsoft/winget-cli/pulls/2919)

### CODE REVIEW

**Review by JohnMcPMS — APPROVED** [review]

**Review by AmelBawa-msft — APPROVED** [review]

File: src/Microsoft.Management.Deployment/PackageVersionInfo.cpp [hunk 1]
```diff
@@ -138,4 +139,24 @@ namespace winrt::Microsoft::Management::Deployment::implementation
...
+
+        if (result.installer.has_value())
+        {
+            auto packageInstallerInfo = winrt::make_self<wil::details::module_count_wrapper<winrt::Microsoft::Management::Deployment::implementation::PackageInstallerInfo>>();
+            packageInstallerInfo->Initialize(result.installer.value());
+            return *packageInstallerInfo;
+        }
+        else
+        {
+            return NULL;
```
Comment (yao-msft) [inline src/Microsoft.Management.Deployment/PackageVersionInfo.cpp:159]:
```suggestion
            return nullptr;

```

File: src/AppInstallerCLIE2ETests/Interop/InstallInterop.cs [hunk 2]
```diff
@@ -12,6 +12,7 @@ namespace AppInstallerCLIE2ETests.Interop
     using Microsoft.Management.Deployment;
     using Microsoft.Management.Deployment.Projection;
     using NUnit.Framework;
+    using Windows.System;
```
Comment (yao-msft) [inline src/AppInstallerCLIE2ETests/Interop/InstallInterop.cs:15]: nit: is this actually used?

File: src/Microsoft.Management.Deployment/PackageManager.idl [hunk 3]
```diff
@@ -280,6 +280,13 @@ namespace Microsoft.Management.Deployment
         /// a lot of fields and no one requesting it.
         /// Gets the manifest of this package version.
         /// virtual Manifest::Manifest GetManifest() = 0;
+    
+        [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 6)]
+        {
+            /// Gets the applicable installer for this package version.
+            PackageInstallerInfo GetApplicableInstaller(InstallOptions options);
+
```
Comment (yao-msft) [inline src/Microsoft.Management.Deployment/PackageManager.idl:288]: nit: empty line in the end

File: src/Microsoft.Management.Deployment/PackageManager.idl [hunk 4]
```diff
@@ -280,6 +280,13 @@ namespace Microsoft.Management.Deployment
         /// a lot of fields and no one requesting it.
```
Comment (yao-msft) [inline src/Microsoft.Management.Deployment/PackageManager.idl:280]: nit: Comments related to GetManifest() could probably be removed now as we have started returning PackageInstallerInfo and PackageLocales

File: src/Microsoft.Management.Deployment/PackageVersionInfo.cpp [hunk 5]
```diff
@@ -138,4 +139,24 @@ namespace winrt::Microsoft::Management::Deployment::implementation
     { 
         return m_packageVersion; 
     }
+    winrt::Microsoft::Management::Deployment::PackageInstallerInfo PackageVersionInfo::GetApplicableInstaller(InstallOptions options)
```
Comment (yao-msft) [inline src/Microsoft.Management.Deployment/PackageVersionInfo.cpp:142]: Nit: let's move this method above GetRepositoryPackageVersion() to let interface methods stay together. And internal methods stay below.

**Review by yao-msft — APPROVED** [review]

**Pull request merged**